nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/lossfunctions/impl/LossSparseMCXENT.java

```diff
@@ -115,6 +118,11 @@ private INDArray toOneHot(INDArray labels, INDArray preOutput){
         return oneHotLabels;
     }
 
+    @Override
+    public SDVariable defineLossArray(@NonNull SameDiff sameDiff, @NonNull SDVariable input,
+            @NonNull SDVariable labels) {
+        throw new UnsupportedOperationException("toSameDiff for LossSparseMCXENT is not yet supported.");
```

Comment:
SameDiff.oneHot op + same defineLossArray as MCXENT would be the quick (albeit memory inefficient) way to support this.

Reply:
It doesn't look like there's any way to pass `depth` a SDVariable, which would be necessary to match the DL4J implementation `OneHot(labels.reshape(labels.length()), oneHotLabels, (int)preOutput.size(-1))`.